src/poetry/packages/locker.py

```diff
@@ -206,175 +197,6 @@ def locked_repository(self) -> Repository:
-            requirement = dependencies.pop(0)
-            if (requirement, requirement.marker) in visited:
-                continue
-            visited.add((requirement, requirement.marker))
-
-            locked_package = cls.__get_locked_package(
-                requirement, packages_by_name, nested_dependencies
-            )
-
-            if not locked_package:
-                raise RuntimeError(f"Dependency walk failed at {requirement}")
-
-            if requirement.extras:
-                locked_package = locked_package.with_features(requirement.extras)
-
-            # create dependency from locked package to retain dependency metadata
-            # if this is not done, we can end-up with incorrect nested dependencies
-            constraint = requirement.constraint
-            marker = requirement.marker
-            requirement = locked_package.to_dependency()
-            requirement.marker = requirement.marker.intersect(marker)
-
-            requirement.constraint = constraint
-
-            for require in locked_package.requires:
-                if require.is_optional() and not any(
-                    require in locked_package.extras[feature]
-                    for feature in locked_package.features
-                ):
-                    continue
-
-                require = deepcopy(require)
-                require.marker = require.marker.intersect(
-                    requirement.marker.without_extras()
-                )
-                if not require.marker.is_empty():
-                    dependencies.append(require)
-
-            key = locked_package
-            if key not in nested_dependencies:
-                nested_dependencies[key] = requirement
-            else:
-                nested_dependencies[key].marker = nested_dependencies[key].marker.union(
-                    requirement.marker
-                )
-
-        return nested_dependencies
-
-    @classmethod
-    def get_project_dependencies(
-        cls,
-        project_requires: list[Dependency],
-        locked_packages: list[Package],
-    ) -> Iterable[tuple[Package, Dependency]]:
-        # group packages entries by name, this is required because requirement might use
-        # different constraints.
-        packages_by_name: dict[str, list[Package]] = {}
-        for pkg in locked_packages:
-            if pkg.name not in packages_by_name:
-                packages_by_name[pkg.name] = []
-            packages_by_name[pkg.name].append(pkg)
-
-        # Put higher versions first so that we prefer them.
-        for packages in packages_by_name.values():
-            packages.sort(
-                key=lambda package: package.version,
-                reverse=True,
-            )
-
-        nested_dependencies = cls.__walk_dependencies(
-            dependencies=project_requires,
-            packages_by_name=packages_by_name,
-        )
-
-        return nested_dependencies.items()
-
-    def get_project_dependency_packages(
```

Comment:
A quick GitHub search shows that this is at least used by one popular project: https://github.com/cloud-custodian/cloud-custodian/blob/318f2b4b1c81b4ffae3474c92038ad328ce00dab/tools/dev/poetrypkg.py#L221-L223.
Not that this holds us from removing this method, but we should probably mention this removal in 1.2 release notes.

Reply:
Good point. I agree we should mention that the functionality has been moved to `poetry-plugin-export`.

I was just thinking about whether we should deprecate the methods and call the functions in the plugin but I don't think that's necessary since we moved the complete export code between 1.1 and 1.2 so that can't be imported as before either. Further, calling export code here is counterproductive for disentangling `poetry` and `poetry-plugin-export`. Thus, a notice in the changelog must suffice.